### PR TITLE
Feature/oriented points layer

### PR DESCRIPTION
**Pull request by seankmartin**
Adds an oriented point layer JSON generator, as well as an encoder for oriented point data as a set of three lines for each point.

The three lines would represent the rotated x, y, and z vectors for each point.

Along with adding this feature, some base annotation encoders and point annotation shader classes that can be used by child classes have been added. This is in an effort to make changes to the overall annotation strategy or shaders easier to propagate through the case base.

### CODE REVIEW

File: cryoet_data_portal_neuroglancer/models/shader_builder.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
+"""Create GLSL shaders for Neuroglancer."""
+
+from typing import Any, Iterable, Optional
+
+TAB = "  "
+
+
+class ShaderBuilder:
```
Comment (seankmartin) [inline cryoet_data_portal_neuroglancer/models/shader_builder.py:8]: Note: the overall shader builder setup will be removed from this PR when the other PR with this builder is merged to main

File: cryoet_data_portal_neuroglancer/precompute/sharding.py
Comment (aranega) [inline cryoet_data_portal_neuroglancer/precompute/sharding.py:1]: There is a dependency to CloudVolume, as this code is a copy, more or less, from the CloudVolume version, perhaps we could try to remove it from our repo and rely on the one of CloudVolume, what do you think?

**Review by aranega — APPROVED** [review]
All looks good to me :) there is only the question about the "sharding.py" file that we could remove probably if we use the CloudVolume library (which is now a dependency of the project).

File: cryoet_data_portal_neuroglancer/precompute/sharding.py
Comment (seankmartin) [reply cryoet_data_portal_neuroglancer/precompute/sharding.py:1]: Thanks for the review! I think this idea sounds good, I will try it and let you know if I run into issues

**Comment by seankmartin** [comment]
Hey @aranega , depending on cloud-volume sharding seems to have worked, you can see the changes in https://github.com/chanzuckerberg/cryoet-data-portal-neuroglancer/pull/13/commits/b34bb95c729cbaafe8a24523ac86d049adde7b9f

File: cryoet_data_portal_neuroglancer/models/json_generator.py [hunk 2]
```diff
@@ -141,33 +148,40 @@ def __post_init__(self):
...
+    line_width: float = 1.0
+
+    def _get_shader(self):
+        shader_builder = OrientedPointShaderBuilder(
+            point_size_multiplier=self.point_size_multiplier,
+            is_instance_segmentation=self.is_instance_segmentation,
+            color=self.color,
+            line_width=self.line_width,
+        )
+        return shader_builder.build_shader()
```
Comment (seankmartin) [inline cryoet_data_portal_neuroglancer/models/json_generator.py:182]: .build

File: cryoet_data_portal_neuroglancer/models/json_generator.py [hunk 3]
```diff
@@ -141,33 +148,40 @@ def __post_init__(self):
...
-            f"  setColor(vec4({set_color}, opacity));\n"
-            f"  setPointMarkerSize(pointScale * prop_diameter());\n"
-            f"  setPointMarkerBorderWidth(0.1);\n"
-            f"}}"
+        shader_builder = NonOrientedPointShaderBuilder(
+            point_size_multiplier=self.point_size_multiplier,
+            is_instance_segmentation=self.is_instance_segmentation,
+            color=self.color,
         )
+        return shader_builder.build_shader()
```
Comment (seankmartin) [inline cryoet_data_portal_neuroglancer/models/json_generator.py:156]: .build

File: cryoet_data_portal_neuroglancer/precompute/points.py [hunk 4]
```diff
@@ -1,95 +1,121 @@
...
+                    enum_values=list(self.names_by_id.keys()),
+                    enum_labels=list(self.names_by_id.values()),
+                ),
+                AnnotationPropertySpec(id="diameter", type="float32"),
+                AnnotationPropertySpec(id="point_index", type="float32"),
+                AnnotationPropertySpec(id="color", type="rgb"),
+            ],
+        )
+
+    def process_data_to_writer_specifications(self, data: list[dict[str, Any]], metadata: dict[str, Any]):
```
Comment (seankmartin) [inline cryoet_data_portal_neuroglancer/precompute/points.py:107]: format

**Pull request merged**